spec/models/manageiq/providers/foreman/provider_spec.rb

```diff
@@ -69,4 +75,28 @@
       expect(provider.provisioning_manager.name).to eq('New Name Provisioning Manager')
     end
   end
+
+  context "ensure_managers callback" do
+    it "automatically creates a provisioning and configuration manager if none is provided" do
+      provider = FactoryBot.create(:provider_foreman)
+      expect(provider.provisioning_manager).to be_kind_of(ManageIQ::Providers::Foreman::ProvisioningManager)
+      expect(provider.configuration_manager).to be_kind_of(ManageIQ::Providers::Foreman::ConfigurationManager)
+    end
+
+    it "sets the provisioning and configuration manager to disabled if created in the maintenance zone" do
+      provider = FactoryBot.create(:provider_foreman, :zone => Zone.maintenance_zone)
+      expect(provider.provisioning_manager.enabled).to eql(false)
```

Comment:
Prefer `eq` or `be` over `eql`.  The reason is that `eq` and `be` is more like `==` whereas `eql` is more like `Object#eql?` meaning it's a hash match.  In this case it's not much different but for consistency we use `eq` everywhere else.

Reply:
Ok, updated.